Simulation/app/simulator.py

```diff
@@ -56,46 +58,50 @@ def main(COMPort = 'COM7', host = 'localhost', port = 2000):
     Port - Port on the host to connect to (Default = 2000)
     """
 
-    #Set logging level
-    logging.basicConfig(format='%(levelname)s: %(message)s', level=logging.INFO)
+    # Set logging level
+    logging.basicConfig(
+        format='%(levelname)s: %(message)s', level=logging.INFO)
     logging.info('listening to server %s:%s', host, port)
 
-    #Initialize some important variables
+    # Initialize some important variables
     client = None
     controller = None
 
     try:
-        #Create client object to interact with server
+        # Create client object to interact with server
         client = Client(host, port)
         client.connectToVehicle()
+        print('success')
 
-        #Define the controller of the vehicle
-        #Intend to add the option for manual control here later with a given param
-        controller = Elcano.RouterboardInterface(COMPort, client.vehicle)
+        # Define the controller of the vehicle
+        # Intend to add the option for manual control here later with a given param
+        if select_controller == 'custom_controller':
+            controller = custom_controller.Autopilot(client.vehicle, False)
+        else:
+            controller = Elcano.RouterboardInterface(COMPort, client.vehicle)
 
-        #Give it one second to catch up, I find without this sometimes I will run into issues
+        # Give it one second to catch up, I find without this sometimes I will run into issues
         time.sleep(1)
 
-        #Continously render
+        # Continously render
         while True:
             client.clock.tick_busy_loop(60)
             client.render()
-
-
-
-    finally: 
-        #Once we are done, destroy the client and controller
+            try:
+                controller.mainloop(client.getKeys(), client.states)
+            except:
+                print("Unexpected error:", sys.exc_info())
```

Comment:
This line will keep raising exceptions when using the routerboardInterface, needs to be changed.